python/feldera/rest/client.py

```diff
@@ -0,0 +1,455 @@
+        )
+
+        connector.id = resp.get("connector_id")
+
+    def delete_connector(self, name: str):
+        """
+        Deletes a connector by name
+        :param name: The name of the connector
+        """
+        self.http.delete(
+            path=f"/connectors/{name}",
+        )
+
+    @staticmethod
+    def __pipeline_from_dict(pipeline: dict) -> Pipeline:
+        attached_connectors = pipeline.get("attached_connectors")
+        descriptor = pipeline.get("descriptor")
+        return Pipeline(
+            name=descriptor.get("name"),
+            program_name=descriptor.get("program_name"),
+            version=descriptor.get("version"),
+            description=descriptor.get("description"),
+            id=descriptor.get("id"),
+            config=descriptor.get("config"),
+            state=pipeline.get("state"),
+            attached_connectors=[
+                AttachedConnector(
+                    connector_name=con.get("connector_name"),
+                    is_input=con.get("is_input"),
+                    relation_name=con.get("relation_name"),
+                    name=con.get("name"),
+                )
+                for con in attached_connectors
+            ] if attached_connectors else []
+        )
+
+    def pipelines(self) -> list[Pipeline]:
+        """
+        Get all pipelines
+        """
+
+        resp = self.http.get(
+            path="/pipelines",
+        )
+
+        return [self.__pipeline_from_dict(pipeline) for pipeline in resp]
+
+    def get_pipeline(self, name: str) -> Pipeline:
+        """
+        Get a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}",
+        )
+
+        return self.__pipeline_from_dict(resp)
+
+    def create_pipeline(self, pipeline: Pipeline):
+        """
+        Create a pipeline
+        """
+
+        body = {
+            "config": pipeline.config,
+            "description": pipeline.description or "",
+            "connectors": pipeline.attached_connectors,
+            "program_name": pipeline.program_name,
+        }
+
+        resp = self.http.put(
+            path=f"/pipelines/{pipeline.name}",
+            body=body,
+        )
+
+        pipeline.id = resp.get("pipeline_id")
+
+    def get_pipeline_config(self, name: str) -> dict:
+        """
+        Get the configuration of a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/config",
+        )
+
+        return resp
+
+    def validate_pipeline(self, name: str) -> bool:
+        """
+        Validate a pipeline.
+        Checks whether the pipeline is configured correctly.
+        This includes checking whether the pipeline references a valid compiled program,
+        whether the connectors reference valid tables/views in the program, and more.
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/validate",
+        )
+
+        return "success" in resp

```

Comment:
Should this not use the `status_code`?

Reply:
If the status code is 400-600, it doesn't reach here, as `_httprequests.__validate` calls `raise_for_status()`.